[0.19] Backports

fanquake opened the pull request:
Backports the following to the 0.19 branch:

* #17946 - Fix GBT: Restore "!segwit" and "csv" to "rules" key
* #18160 - gui: Avoid Wallet::GetBalance in WalletModel::pollBalanceChanged
* #18425 - releases: Update with new Windows code signing certificate
* #18676 - build: Check libevent minimum version in configure script
* #19097 - qt: Add missing QPainterPath include (as per #19510)
* #18640 - appveyor: Remove clcache
* #19444 - test: Remove cached directories and associated script blocks from appveyor config
* #19612 - lint: fix shellcheck URL in CI install
* #18001 -  Updated appveyor job to checkout a specific vcpkg commit ID

Closes: #19510.

luke-jr committed:
Bugfix: Include "csv","!segwit" in "rules"

They have been missing since buried deployments were merged

Github-Pull: #17946
Rebased-From: 2abe8cc3b760219cfa434e4c96e9f8d3611d0037

luke-jr committed:
QA: feature_segwit: Check that template "rules" includes "!segwit" as appropriate

Github-Pull: #17946
Rebased-From : 412d5fe8791c417bf46fc55a5bb8d59be98a33db

sipsorcery committed:
Updated appveyor job to checkout a specific vcpkg commit ID.

Github-Pull: #18001
Rebased-From: 2bcc70531ac88e29a7d59a1ab8c787c55e7f88f5

promag committed:
gui: Avoid Wallet::GetBalance in WalletModel::pollBalanceChanged

Github-Pull: #18160
Rebased-From: 0933a37078e1ce3a3d70983c3e7f4b3ac6c3fa37

achow101 committed:
Update with new Windows code signing certificate

Github-Pull: #18425
Rebased-From: 3e0df92bf216e1dce05ca9bf14049f2e42783c30

Commit:
appveyor: Remove clcache

Github-Pull: #18640
Rebased-From: fac0c8db9f634ea7231fb0b6dc139ae0bda4a051

hebasto committed:
build: Set libevent minimum version to 2.0.21

Github-Pull: #18676
Rebased-From: b68e71796792a9da9daa0a4e759d284d15595230

promag commented:
Do you want to include #18542 here?

achow101 committed:
Add missing QPainterPath include

This is needed to compile with Qt 5.15.

Github-Pull: #19097
Rebased-From: 79b0a69e09c1a912122e6431ea3c530cc292c690

fanquake commented:
> Do you want to include #18542 here?

Thanks. I've pulled in d3cfe0d6a472c748142b96c33de866478cca5545 from that PR.

luke-jr commented:
#19185 is a clean merge of the original commit. No need for an ugly re-commit as here.

Should remove the commits here, and reopen #19185 instead

luke-jr reviewed (changes requested):
#19185 is a clean merge of the original commit. No need for an ugly re-commit as here.

Should remove the commits here, and reopen #19185 instead

promag commented:
#19300 was merged.

fanquake committed:
test: remove Cirrus CI FreeBSD job

This has been removed from the master branch, and always seems to fail
at the installing packages phase. i.e:
```bash
pkg install -y autoconf automake boost-libs git gmake libevent libtool pkgconf python3 ccache
Updating FreeBSD repository catalogue...
Fetching meta.txz: . done
pkg: repository meta /var/db/pkg/FreeBSD.meta has wrong version 2
repository FreeBSD has no meta file, using default settings
Fetching packagesite.txz: .......... done
pkg: repository meta /var/db/pkg/FreeBSD.meta has wrong version 2
pkg: Repository FreeBSD load error: meta cannot be loaded No error: 0
Unable to open created repository FreeBSD
Unable to update repository FreeBSD
Error updating repositories!

Exit status: 3
```

Github-Pull: #19224
Rebased-From: e7f06f9b0e84a65812d24ff6efa4bc2d3d818590

sipsorcery committed:
Remove cached directories and associated script blocks from appveyor CI configuration.

Github-Pull: #19444
Rebased-From: 961e667600441c35845fcb36b120500c813cd3ed

fanquake committed:
lint: fix shellcheck URL in CI install

This is causing the tests to fail for backports etc.

Github-Pull: #19612
Rebased-From: 7ee4769cd4be0975254b218134d057c2bb443509

jnewbery commented:
ACK be951470bea132b4cbe1823cd564bf14aaf0ea7e

MarcoFalke commented:
cherry-pick ACK be951470bea132b4cbe1823cd564bf14aaf0ea7e 🌎

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

cherry-pick ACK be951470bea132b4cbe1823cd564bf14aaf0ea7e 🌎
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhiqAv+LLwsV4mNGsaTSOa3q9mT14B/bWuib9+CFffg8lBvedHfjs/B/jRRkbDt
R8zCFOZPJHed/8M3ICUYHi04FJkO5y4Bh8TzIY8087SPM6sDatE/MWXSCe5f9Ttt
aFTfYx19As/yaE4cMJZD+/3pwHVObZuQ4nzu9TZYbtUBRoJj912QP5Yq+gBZXxwt
M10O6pwz7e5IXwGCPeIIlhMLgiqmom7pJIsijZi3PaGeH+oR8sBnxB99B4dVwiJl
+E7VQSVldHNjoNcSftvp41deDc1L79hFn3au4ws2Zq6y1BXlgAxLOqBmkGSS6y1H
Qvq+YrtQM/pGblgRuXUD3Nb3qbehk7mVPka+RlI3diyj1yB6YbG0YAjDxvDmkQgp
VBqW7loKrAnxO6VRpaZ8+SGZyWwCQQQEii25WKvRZm4xR/5AoJgHv4PU1hK89jei
lzayyDZVUqbWS8PDD4zhes28m1u5iZaTKgALYFegyxnRl5YHuwSfACuLjQT2CMYM
wpMzwC5O
=AqoC
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `85457e8568a50d11680719483b91ec171c3e108883edca3d38bbae169190b69a  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e89294010885457e8568a50d11680719483b91ec171c3e108883edca3d38bbae169190b69af0108007f04033f96fa8ccd06c1c66fd39ca08fff01031cdb7efbba19611fa452b7321e3b83608f1045f325fbff008fb74f240498991840083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010b8be27e7bc024f17e93377cc6df037ec08f1045f325fbef0084dd6694a334568d00083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df010cc9e343415e3f0a88fa1784e1540b1ec08f1045f325fbef00885be8dc39a313b990083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

MarcoFalke commented on .appveyor.yml:
```diff
@@ -5,78 +5,54 @@ configuration: Release
+      cd c:\tools\vcpkg
+      $env:GIT_REDIRECT_STDERR = '2>&1' # git is writing non-errors to STDERR when doing git pull. Send to STDOUT instead.
+      git pull origin master > $null
+      git checkout $env:VCPKG_COMMIT_ID
```
Any reason to check out the same commit twice?

You can avoid the conflicts by cherry-picking in the correct order:

```
$ git log -3 --oneline 
a51ffbe92c (HEAD) Remove cached directories and associated script blocks from appveyor CI configuration.
6646b5f891 appveyor: Remove clcache
bd95892b0d Updated appveyor job to checkout a specific vcpkg commit ID.

MarcoFalke commented:
> #19300 was merged.

This wasn't included here, but I went ahead and merged this to make the ci green on the other backport pulls
